pkg/controller/constraint/constraint_controller.go

```diff
@@ -344,21 +344,24 @@ func (r *ReconcileConstraint) Reconcile(ctx context.Context, request reconcile.R
 						return reconcile.Result{}, err
 					}
 					hasVAP, err := ShouldGenerateVAP(unversionedCT)
-					if err != nil {
+					switch {
+					case errors.Is(err, celSchema.ErrCodeNotDefined):
+						generateVAPB = false
```

Comment:
do we want to log warning here?

`log.V(1).Info("Warning: Template is not configured to create ValidatingAadmissionPolicy, cannot create ValidatingAdmissionPolicyBinding")`

would this be leaking information intended for template to constraints?

Reply:
IMO we should log and update status ONLY if code.engine is K8sNativeValidation. Otherwise this error will appear for all older rego CTs as code.source was introduced recently.

e.g. of older regos CTs:

```yaml
targets:
    - target: admission.k8s.gatekeeper.sh
      rego: |
        package k8sallowedrepos
```

Reply:
I don't see changes that address 
> IMO we should log and update status ONLY if code.engine is K8sNativeValidation.

This would require updating framework code I think. https://github.com/open-policy-agent/frameworks/blob/aa99306df54ee1e75380507503a547f22686b11d/constraint/pkg/client/drivers/k8scel/schema/schema.go#L291 to only return this error when `code.engine == Name`

This isnt blocking and can be added as a followup.